htdp-lib/2htdp/private/image-more.rkt

```diff
@@ -1130,17 +1131,23 @@
   (mk-text string font-size color face family style weight underline))
 
 (define (mk-text str font-size color face family style weight underline)
-  (cond
-    [(<= (string-length str) 1)
-     (mk-single-text str font-size color face family style weight underline)]
-    [else
-     (let ([letters (string->list str)])
-       (beside/internal
-        'baseline
-        (mk-single-text (string (car letters)) font-size color face family style weight underline)
-        (map (λ (letter)
-               (mk-single-text (string letter) font-size color face family style weight underline))
-             (cdr letters))))]))
+  (let ([lines (string-split str "\n")])
+       (apply
+        above/align
+        "left"
```

Comment:
Of course, this right here is begging to be parameterized to allow different justification options.